### PR TITLE
Refactored Code

### PR DESCRIPTION
**Pull request by Kwaleyela-Ikafa**
In this branch we:

- Followed YAGNI, KISS, and DRY rules
- Followed SOLID principles
- The role of main.rb is only to start the code and ask the user for the initial choice of options

### CODE REVIEW

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @Kwaleyela-Ikafa  , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [x] Data is not being saved. Kindly fix it
image


_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kwaleyela-Ikafa** [comment]
@m05e5 you were supposed to make the file executable by adding  [   #!/usr/bin/env ruby   ] to the top of the main.rb file and change the file permissions to execute using [ chmod 755 main.rb ]  only then would you be able to run the script in the terminal by typing ./main.rb and the app would not exit and you would have access to the data. saving data is not part of the project requirements for week 4 day 1

**Comment by m05e5** [comment]
@Kwaleyela-Ikafa  Sorry my bad. can you ask for another review I'm sure the next reviewer will understand. Sorry again

**Comment by Kwaleyela-Ikafa** [comment]
@m05e5 it's okay, I'll add a loop to it so the next reviewer can use it in vs code

File: create_rental.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @people.each_with_index do |person, index|
+      puts "ID:#{index + 1}, Name: #{person.name}, Age: #{person.age}"
+    end
+    person_id = gets.chomp.to_i
+    puts 'Person selected'
+
+    print 'Date (yyyy-mm-dd): '
+    date = gets.chomp.to_s
+    puts 'Date of renting book is added'
+    @rentals << Rental.new(date, @people[person_id - 1], @books[book_id - 1])
```
Comment (ShahierNashaat) [inline create_rental.rb]: - [x] Kindly notice that you are swapping the arguments when you are creating the `rental` and this causing error please fix that.

```ruby
@rentals << Rental.new(date, @books[book_id - 1], @people[person_id - 1])
```
Capture

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Kwaleyela-Ikafa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: create_rental.rb (hunk 1 above)
Comment (Kwaleyela-Ikafa) [reply create_rental.rb]: Thank you @ShahierNashaat I couldn't find the source of the error and I submitted in the hopes that a reviewer would see it and show me the fix lol

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- I see from your comment that you were struggling to find the solution to the issue the previous CR raised, good job figuring out how to solve it on your own! 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**